kolibri/core/assets/src/core-app/baseClient.js

```diff
@@ -4,11 +4,43 @@
  * circular dependencies and repeated code.
  */
 import rest from 'rest';
-import csrf from 'rest/interceptor/csrf';
+import interceptor from 'rest/interceptor';
 import errorCode from 'rest/interceptor/errorCode';
 import cookiejs from 'js-cookie';
 
-export default rest.wrap(errorCode).wrap(csrf, {
-  name: 'X-CSRFToken',
-  token: cookiejs.get('csrftoken'),
+/*
+ * Vendored and modified from https://github.com/cujojs/rest/blob/master/interceptor/csrf.js
+ */
+
+/**
+ * Applies a Cross-Site Request Forgery protection header to a request
+ *
+ * CSRF protection helps a server verify that a request came from a
+ * trusted  client and not another client that was able to masquerade
+ * as an authorized client. Sites that use cookie based authentication
+ * are particularly vulnerable to request forgeries without extra
+ * protection.
+ *
+ * @see http://en.wikipedia.org/wiki/Cross-site_request_forgery
+ *
+ * @param {Client} [client] client to wrap
+ *
+ * @returns {Client}
+ */
+const csrf = interceptor({
+  request: function handleRequest(request) {
+    var headers, name, token;
+
+    headers = request.headers || (request.headers = {});
+    name = request.csrfTokenName || 'X-CSRFToken';
+    token = request.csrfToken || cookiejs.get('csrftoken');
```

Comment:
It's not clear where these names are coming from - I assume we have control over them on the server-side?

If we know the names of the header and the cookie ahead of time, I would expect the `||` logic to be unnecessary, and the values simply hard-coded here, perhaps with a comment indicating what middleware sets them on the server.

Reply:
This is vendored from our rest client, I did briefly simplify this code, but it bought very little, and it seemed safer just to bring in the code as is.

Reply:
cool. worth maybe just adding a comment indicating where on the backend these are handled?

Reply:
They aren't anywhere in our backend code, it's all in Django magic. Can link to the Django CSRF docs if helpful.